doc-jsdoc/pentaho/context/spec/IVariables.jsdoc

```diff
@@ -0,0 +1,119 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * The `spec.IVariables` interface describes the variables that
+ * define a Pentaho client-side context.
+ *
+ * A `null` variable value is possible and means that the variable is not defined or fixed.
+ *
+ * @name pentaho.context.spec.IVariables
+ * @interface
+ */
+
+/**
+ * The base path of the Pentaho web application (e.g. `"/pentaho/"`).
+ *
+ * In embedded scenarios,
+ * this property may have the same value as {@link pentaho.context.spec.IVariables#serverUrl}.
+ * Otherwise, it contains an absolute virtual path for the current web server.
+ *
+ * @name basePath
+ * @memberOf pentaho.context.spec.IVariables#
+ * @type {?nonEmptyString}
+ */
+
+/**
+ * The identifier of the client application.
+ *
+ * This is the value specified in a web context script URL, as in: `webcontext.js?context=application-id`.
```

Comment:
I've come to agree with @graimundo on keeping the two concepts separate. The main/default context will never have an application defined. Applications must create a context where they override its `application` id property. They currently already do this when creating their corresponding `pentaho/type/Context`.